src/helpers.ts

```diff
@@ -157,6 +163,63 @@ function getNativeTokenName(chainId: BigInt): string {
   }
 }
 
+export function createSnapshot(timestamp: BigInt): Snapshot {
```

Comment:
Maybe this name would be more precise.
```suggestion
export function getOrCreateSnapshot(timestamp: BigInt): Snapshot {
```

Reply:
sound great

Reply:
I think it would be great to also change `createVolumeSnapshot`